### PR TITLE
feature/here

### PR DESCRIPTION
**Pull request by wolter**
Custom Module for Here

### CODE REVIEW

File: modules/here/README.md [hunk 1]
```diff
@@ -0,0 +1,204 @@
+Adds [**Here**](https://developer.here.com/) specific features as Flow Nodes to Cognigy.AI. Please note, you need an APP ID and an APP CODE for the **Here** API which you can generate for free with your **Here** developer account [here](https://developer.here.com/develop/rest-apis?create=Freemium-Basic).
+
```
Comment (mastasky) [inline modules/here/README.md]: Please provide a proper description of the needed Secret. See this example: https://github.com/Cognigy/CustomModules/tree/master/modules/hubspot

File: modules/here/src/module.ts [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+ *
+ * @arg {SecretSelect} `secret` Secret with app_id and app_code
+ * @arg {String} `searchtext` Query to turn into a geolocation
+ * @arg {Boolean} `writeToContext` Whether to write to Cognigy Context (true) or Input (false)
+ * @arg {CognigyScript} `store` Where to store the result
+ * @arg {Boolean} `stopOnError` Whether to stop on error or continue
+ *
+ */
+async function geocoderGeocode(input: IFlowInput, args: { secret: CognigySecret, searchtext: string, writeToContext: boolean, store: string, stopOnError: boolean }): Promise<IFlowInput | {}> {
+
```
Comment (mastasky) [inline modules/here/src/module.ts]: Please check if the required parameters are present in the secret and if not, return a rejected Promise with an error.

`if (!args.secret || !args.secret.app_id || !args.secret.app_code) return Promise.reject("Secret not defined or invalid.");`

File: modules/here/src/module.ts [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    try {
```
Comment (mastasky) [inline modules/here/src/module.ts]: Please add checks for all required parameters (in this case args.searchText)

File: modules/here/src/module.ts [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+ * @arg {Number} `latitude` Latitude of the geolocation
+ * @arg {Number} `longitude` Longitude of the geolocation
+ * @arg {String} `searchtext` Query to turn into a geolocation
+ * @arg {Boolean} `writeToContext` Whether to write to Cognigy Context (true) or Input (false)
+ * @arg {CognigyScript} `store` Where to store the result
+ * @arg {Boolean} `stopOnError` Whether to stop on error or continue
+ *
+ */
+async function placesDiscoverSearch(input: IFlowInput, args: { secret: CognigySecret, latitude: number, longitude: number, searchtext: string, writeToContext: boolean, store: string, stopOnError: boolean }): Promise<IFlowInput | {}> {
+
```
Comment (mastasky) [inline modules/here/src/module.ts]: See above for secret and parameter checks

File: modules/here/src/module.ts [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+ * @arg {Number} `zoom` Zoom level for the image
+ * @arg {Number} `width` Width for the image
+ * @arg {Number} `height` Height for the image
+ * @arg {Boolean} `writeToContext` Whether to write to Cognigy Context (true) or Input (false)
+ * @arg {CognigyScript} `store` Where to store the result
+ * @arg {Boolean} `stopOnError` Whether to stop on error or continue
+ *
+ */
+async function mapsImageMapview(input: IFlowInput, args: { secret: CognigySecret, latitude: number, longitude: number, zoom: number, width: number, height: number, writeToContext: boolean, store: string, stopOnError: boolean }): Promise<IFlowInput | {}> {
+
```
Comment (mastasky) [inline modules/here/src/module.ts]: See above for secret and parameter checks

File: modules/here/README.md (hunk 1 above)
Comment (wolter) [reply modules/here/README.md]: done

File: modules/here/src/module.ts (hunk 2 above)
Comment (wolter) [reply modules/here/src/module.ts]: done

File: modules/here/src/module.ts (hunk 3 above)
Comment (wolter) [reply modules/here/src/module.ts]: done

Comment (wolter) [reply modules/here/src/module.ts]: done

File: modules/here/src/module.ts (hunk 5 above)
Comment (wolter) [reply modules/here/src/module.ts]: done

File: modules/here/README.md (hunk 1 above)
Comment (mastasky) [reply modules/here/README.md]: Please provide a bullet list of the needed parameters, like in the module link above

**Comment by mastasky** [comment]
@monty0157 - someone needs to do a functional review.

**Pull request merged**